connexion/middleware/validation.py

```diff
@@ -0,0 +1,232 @@
+"""
+Validation Middleware.
+"""
+import logging
+import pathlib
+import typing as t
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+from connexion.apis.abstract import AbstractSpecAPI
+from connexion.decorators.uri_parsing import AbstractURIParser
+from connexion.exceptions import MissingMiddleware, UnsupportedMediaTypeProblem
+from connexion.http_facts import METHODS
+from connexion.middleware import AppMiddleware
+from connexion.middleware.routing import ROUTING_CONTEXT
+from connexion.operations import AbstractOperation
+from connexion.resolver import ResolverError
+from connexion.utils import is_nullable
+from connexion.validators import JSONBodyValidator
+
+from ..decorators.response import ResponseValidator
+from ..decorators.validation import ParameterValidator
+
+logger = logging.getLogger("connexion.middleware.validation")
+
+VALIDATOR_MAP = {
+    "parameter": ParameterValidator,
+    "body": {"application/json": JSONBodyValidator},
```

Comment:
I have been struggling here with the distinction of `parameter` and `body` as there currently isn't a clear split: `form` parameters are validated by the `ParameterValidator` but also by the `RequestBodyValidator`. The request body also contains parameters, at least in Swagger 2 ([link](https://swagger.io/docs/specification/2-0/describing-parameters/#types)). In OpenAPI there is a clearer distinction between "parameters" and the "body" ([link](https://swagger.io/docs/specification/describing-parameters/#types)).

Seeing this, I think a way forward could be to move everything of formdata to a separate body validator as well. This also simplifies things wrt accessing the ASGI scope only vs also needing the `send` and `receive` callables. Wdyt?

Reply:
Yes, I think that makes sense. Just let me rephrase to make sure I understand:

- Move form validation into a separate body validator so it has the full ASGI interface (scope, send, receive) available and can decide if / how to consume the stream
- Parameter validators will only have the scope available, possibly via a Request object which does not provide access to the stream.

Correct?

Reply:
Yes, indeed. This to limit the amount of places in which the body stream could be accessed.

Reply:
👋  does this code mean that the `JSONBodyValidator` is string-literally matched? I ask because I was thinking about a case where an American business is insisting that a partner use their `mm/dd/yyyy` format for dates. They already use Connexion, and I'd sooner specify a separate accept & content-type for their weird expected output, and have everything else use, nice readable, standard-compliant ISO-8601 as per https://datatracker.ietf.org/doc/html/rfc3339#section-5.6 and https://swagger.io/docs/specification/data-models/data-types/#string.